### DISCUSSION TITLE
Bizarre behavior experienced with RRF

### DISCUSSION THREAD

**Discussion by benvrakas** [OrcaSlicer/OrcaSlicer#4426] [General]
Hello, I am running a Fly E3 V3 Pro with the latest RepRap Firmware. I have been battling an issue for the past few days that only happens when I use orca slicer. While printing, my printer will decelerate to 0 and then immediately accelerate back up to printing speed during prints, resulting in random blobs. This only happens at the end of G1-G3 commands - never in the middle of a line segment. The issue works itself out after the first few layers. I have tried using the starting block of gcode from a print generated on prusa slicer that I know works, and swapping in everything after from orca slicer. This still caused the issue to appear even though the gcode was almost indentical. I've included gcode files of identical prints with mostly identical settings, one from PS that works and one from OS that exhibits the bug. I have tried changing extrusion to absolute and many other things to make the profiles identical but that did not fix the issue. I was previously running Bambu Studio on Marlin with some custom Gcode for RRF and it was having the same issues. 
[Files.zip](https://github.com/SoftFever/OrcaSlicer/files/14566896/Files.zip)

**Comment by benvrakas** [comment]
![IMG_0893](https://github.com/SoftFever/OrcaSlicer/assets/24928370/4e162df1-0341-4a6f-a10b-956dbfee3181)
![IMG_0894](https://github.com/SoftFever/OrcaSlicer/assets/24928370/0295a026-bb03-4246-b938-9cc99cc7f334)
With and without

**Reply by Rickthebig** [reply]
I tend to confirm that something is wrong with the Orca/RRF combo. I switched from Cura to Orca a few days ago, did a Pressure Advance test (since I also changed filament) and saw some really weird things: Suddenly retracting about 15 cm (!) and in most times immediately extruding it back, doing stops where unusual large blobs (several mm) are  layed, or just stopping extrusion at all at later phases of the print. Nothing of that could be found in the gcode. It could be the new PA setting (raised from 0.26 to 0.325) are the culprit, but such effects are just irrational. I also need to retest extruder jerk and accelerations, which I have changed in wide ranges without getting a clou.

As I understand, changes in flow due to PA are not put in the gcode by the slicer, but are executed by the board firmware at print time.

**Comment by hliebscher** [comment]
RRF and Orca works very good
i think your rrf start macro is wrong not the start code from orca

my orca start code...
G90 ; use absolute coordinates
M83 ; extruder relative mode
M104 S110
M140 S[first_layer_bed_temperature] ; set bed temp
{if filament_type[initial_tool]=="PC" or filament_type[initial_tool]=="PA"}
M104 S{first_layer_temperature[initial_tool]-25} ; set extruder temp for bed leveling
M109 S{first_layer_temperature[initial_tool]-25} ; wait for temp
{elsif filament_type[initial_tool]=="FLEX"}
M104 S210 ; set extruder temp for bed leveling
M109 S210 ; wait for temp
{else}
M104 S170 ; set extruder temp for bed leveling
M109 S170 ; wait for temp
{endif}
M302 S160 ; lower cold extrusion limit to 160C
{if filament_type[initial_tool]=="FLEX"}
G1 E-4 F2400 ; retraction
{else}
G1 E-2 F2400 ; retraction
{endif}
{if first_layer_bed_temperature[initial_tool]<=60}M106 S100{endif}
M190 S[first_layer_bed_temperature] ; wait for bed temp
M107
M557 X{first_layer_print_min[0]}:{first_layer_print_max[0]} Y{first_layer_print_min[1]}:{first_layer_print_max[1]} S25
G29 S2
G29 S0
; prepare for purge
G0 X0 Y0 F2800
M104 S{first_layer_temperature[0]}
M109 S{first_layer_temperature[0]}
G92 E0
;
; Extrude purge line
;
G92 E0 ; reset extruder position
G1 E{(filament_type[0] == "FLEX" ? 4 : 2)} F2400 ; deretraction after the initial one before nozzle cleaning

G0 E7 X-5 Y-7 Z0.6 F500 ; purge
G0 X5 E4 F500 ; purge
G0 X15 E4 F650 ; purge
G0 X25 E4 F800 ; purge
G0 X{25 + 3} Z{0.05} F{8000} ; wipe, move close to the bed
G0 X{25 + 3 * 2} Z0.25 F{8000} ; wipe, move quickly away from the bed
G92 E0
M221 S100 ; set flow to 100%
M593 P"mzv" F43.6 L0.34
;M593 P"none"      ; disable input shaping
;M592 D0 A0.01 B0.0005 ; set nonlinear parameters for extruder drive 0 



and rrf start macro

T0                                                                          ; ensure the tool is selected
;
M572 D0 S0.0                                                      ; clear pressure advance.
M220 S100                                                          ; set speed factor back to 100% in case it was changed
M221 S100                                                          ; set extrusion factor back to 100% in case it was changed
M290 R0 S0                                                       ; clear any baby-stepping
M106 S0                                                             ; turn part cooling blower off if it is on
M703                                                                  ; execute loaded filament's config.g
M501
G32                                                                    ; execute bed.g (level gantry)
G29 S1                                                               ; load bed mesh for the system's set filament type
M400                                                                 ; finish all moves, clear the buffer
G90                                                                    ; absolute Positioning
M83
